src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -539,71 +504,53 @@ export class SQLFuture extends Disposable implements FutureInternal {
 				metadata: {},
 				execution_count: this._executionCount,
 				data: {
-					'application/vnd.dataresource+json': this.convertToDataResource(resultSet.columnInfo, subsetResult),
-					'text/html': this.convertToHtmlTable(resultSet.columnInfo, subsetResult)
-				}
+					'application/vnd.dataresource+json': this.convertToDataResource(resultSet.columnInfo),
+					'text/html': this.convertToHtmlTable(resultSet.columnInfo)
+				},
+				batchId: resultSet.batchId,
+				id: resultSet.id,
+				queryRunnerUri: this._queryRunner.uri,
 			},
 			metadata: undefined,
 			parent_header: undefined
 		};
 		this.ioHandler.handle(msg);
-		this._querySubsetResultMap.delete(resultSet.id);
 	}
 
-	setIOPubHandler(handler: nb.MessageHandler<nb.IIOPubMessage>): void {
-		this.ioHandler = handler;
-	}
-
-	registerMessageHook(hook: (msg: nb.IIOPubMessage) => boolean | Thenable<boolean>): void {
-		// no-op
-	}
-	removeMessageHook(hook: (msg: nb.IIOPubMessage) => boolean | Thenable<boolean>): void {
-		// no-op
-	}
-
-	private convertToDataResource(columns: IColumn[], subsetResult: ResultSetSubset): IDataResource {
+	private convertToDataResource(columns: IColumn[]): IDataResource {
 		let columnsResources: IDataResourceSchema[] = [];
 		columns.forEach(column => {
 			columnsResources.push({ name: escape(column.columnName) });
 		});
-		let columnsFields: IDataResourceFields = { fields: undefined };
-		columnsFields.fields = columnsResources;
+		let columnsFields: IDataResourceFields = { fields: columnsResources };
 		return {
 			schema: columnsFields,
-			data: subsetResult.rows.map(row => {
-				let rowObject: { [key: string]: any; } = {};
-				row.forEach((val, index) => {
-					rowObject[index] = val.displayValue;
-				});
-				return rowObject;
-			})
+			data: []
 		};
 	}
 
-	private convertToHtmlTable(columns: IColumn[], d: ResultSetSubset): string[] {
-		// Adding 3 for <table>, column title rows, </table>
-		let htmlStringArr: string[] = new Array(d.rowCount + 3);
-		htmlStringArr[0] = '<table>';
-		if (columns.length > 0) {
-			let columnHeaders = '<tr>';
-			for (let column of columns) {
-				columnHeaders += `<th>${escape(column.columnName)}</th>`;
-			}
-			columnHeaders += '</tr>';
-			htmlStringArr[1] = columnHeaders;
+	private convertToHtmlTable(columns: IColumn[]): string[] {
+		let htmlTable: string[] = new Array(3);
+		htmlTable[0] = '<table>';
+		let columnHeaders = '<tr>';
```

Comment:
Please make sure the 'SELECT 1' scenario works where there are no explicit columns defined.